scrunch/datasets.py

```diff
@@ -129,34 +113,23 @@ def get_dataset(dataset, connection=None, editor=False, project=None):
                 "Authenticate first with scrunch.connect() or by providing "
                 "config/environment variables")
     root = connection
-
-    if project:
-        if isinstance(project, six.string_types):
-            project_obj = get_project(project, connection)
-            ds = project_obj.get_dataset(dataset)
-        else:
-            ds = project.get_dataset(dataset)
-    else:
+    # search by dataset name
+    try:
+        shoji_ds = root.datasets.by('name')[dataset].entity
+    except KeyError:
+        # search by dataset id
         try:
-            shoji_ds = root.datasets.by('name')[dataset].entity
+            shoji_ds = root.datasets.by('id')[dataset].entity
         except KeyError:
+            # search by id on any project
```

Comment:
This case builds the url and tries to find the given dataset within any project.